torch/_dynamo/compiled_autograd.py

```diff
@@ -147,6 +151,109 @@ def begin_capture(
         self.stack.enter_context(preserve_node_meta())
         return inputs, sizes, scalars
 
+    def proxy_call_lambda(
+        self,
+        idx,
+        inputs,
+        output_metadatas: List[Optional[Any]],
+    ):
+        with disable_proxy_modes_tracing():
+            # create fake Tensors
+            grad_ins: List[Optional[torch.Tensor]] = []
+            for output_metadata in output_metadatas:
+                if output_metadata is None:
+                    continue
+
+                layout, device, dtype, size = output_metadata
+                grad_ins.append(
+                    torch.empty(size=size, dtype=dtype, layout=layout, device=device)
+                )
+
+            global next_op
+
+            @torch.library.custom_op(  # type: ignore[misc]
+                f"compiled_autograd::cpp_node_op_{next_op}", mutates_args=()
+            )
+            def cpp_node_op_i(
+                inputs: List[torch.Tensor], idx: int
+            ) -> List[torch.Tensor]:
+                print(f"Calling lifted c++ node at idx={idx}")
+                cppouts = torch._C._dynamo.compiled_autograd.call_lambda(inputs, idx)
+                filtered_output_metadatas = [
+                    o for o in output_metadatas if o is not None
+                ]
+                assert len(cppouts) == len(filtered_output_metadatas)
+                # what happens when shit is marked as dynamic...
+                pyouts = []
+                for i, cppout in enumerate(cppouts):
+                    if cppout is None:
+                        layout, device, dtype, size = filtered_output_metadatas[i]
+                        pyouts.append(
+                            torch.empty(
+                                size=size, dtype=dtype, layout=layout, device=device
+                            )
+                        )
+                    else:
+                        # strides could be anything, and can cause issue with
+                        # downstream stride dependent operations
+                        pyouts.append(cppout.clone().contiguous())
```

Comment:
btw, you need a "force_contiguous" here. cpp_out is already assumed to be empty, so the contiguous call will be traced away